### PR TITLE
Developers can add a Balrog Logout button to views with a view helper method

### PR DESCRIPTION
**Pull request by JonathanAndrews**
**Logout**
- view helper method so developers can add a logout button
- method to remove "authentication" from cookie
- redirect to the user to Root

### CODE REVIEW

**Comment by jalada** [comment]
I'd like to tweak the `balrog_logout_button` API so it's a bit more friendly, and a bit more similar to similar Rails methods. Can you make it so it supports the following style of uses:

```erb
<%= balrog_logout_button %>
<%= balrog_logout_button "Leave this place" %>
<%= balrog_logout_button "Click me", class: 'fancy-button' %>
<%= balrog_logout_button class: 'fancy-button--with-default-text' %>
```

**Comment by jalada** [comment]
I suggest some unit tests for all the styles of usage we have in the README, to confirm that they definitely work! (You can pull in the helper method & then check that the HTML matches what you expect)

File: balrog.gemspec [hunk 1]
```diff
@@ -24,6 +24,7 @@ Gem::Specification.new do |spec|
   spec.require_paths = ["lib"]
 
   spec.add_dependency "bcrypt", "~> 3.0"
+  spec.add_dependency "rails"
```
Comment (jalada) [inline balrog.gemspec]: The README says we support >= 5. However I know we're running it on Rails 4.2. I suggest we relax what the README says, and make this `>= 4.2`, as that's the earliest we've tested it against.

**Comment by jalada** [comment]
Once the Rails dependency has been tightened up, please merge and get this out as a new release. @JonathanAndrews what kind of version bump do you think this is?

**Comment by JonathanAndrews** [comment]
@jalada I was thinking a minor bump. From 1.0.0 to 1.1.0?
There have been no bug fixes and all changes are backwards compatible.

**Pull request merged**